### PR TITLE
[Bot-Builder] Add SkillHandlerImpl

### PR DESCRIPTION
**Pull request by Batta32**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        }
+
+        if (withBot == null) {
+            throw new IllegalArgumentException("bot cannot be null");
+        }
+
+        this.skillConversationReferenceKey = withSkillConversationReferenceKey;
+        this.adapter = withAdapter;
+        this.bot = withBot;
+        this.conversationIdFactory = withConversationIdFactory;
```
Comment (matiasroldan6) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:63]: conversationIdFactory and getOAuthScope not being [checked for null](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/Skills/SkillHandlerImpl.cs#L47)?

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        Activity activity) {
+        return processActivity(claimsIdentity, conversationId, activityId, activity);
+    }
+
+    public CompletableFuture<Void> onDeleteActivity(
+        ClaimsIdentity claimsIdentity,
+        String conversationId,
+        String activityId) {
+
+        SkillConversationReference skillConversationReference = getSkillConversationReference(conversationId).join();
```
Comment (matiasroldan6) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:87]: this is quite different from TS so I guess it's only based on C#. Is the join() a good idea?

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+            // We need to use the deprecated method for backward compatibility.
+            ConversationReference conversationReference =
+                this.conversationIdFactory.getConversationReference(conversationId).join();
+            skillConversationReference = new SkillConversationReference();
+            skillConversationReference.setConversationReference(conversationReference);
+            skillConversationReference.setOAuthScope(this.getOAuthScope.get());
+        }
+
+        if (skillConversationReference == null) {
+            if (this.logger != null) {
```
Comment (matiasroldan6) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:177]: didn't know we had to validate the logger for null before 🤔

**Review by matiasroldan6 — APPROVED** [review]

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java (hunk 1 above)
Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:63]: Thanks, fixed!

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java (hunk 3 above)
Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:177]: I think it's not necessary as this is generated in the prop of the object. But, this was a copy/paste from `SkillHandler` as the `Impl` is trying to move things between places

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java (hunk 2 above)
Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:87]: Agree with you, but it was a copy/paste from `SkillHandler`. As you mentioned it's only based on [C#](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/Skills/SkillHandlerImpl.cs#L62-L73) apparently.

I don't think that the `join` is the best approach.

**Pull request merged**